Form/FormFlow.php

```diff
@@ -722,7 +754,16 @@ protected function applyDataFromSavedSteps() {
 	 * {@inheritDoc}
 	 */
 	public function createForm() {
-		return $this->createFormForStep($this->currentStepNumber);
+		$form = $this->createFormForStep($this->currentStepNumber);
+
+		if ($this->expired) {
+			if ($this->hasListeners(FormFlowEvents::FLOW_EXPIRED)) {
+				$event = new FlowExpiredEvent($this, $form);
+				$this->eventDispatcher->dispatch(FormFlowEvents::FLOW_EXPIRED, $event);
+			}
+		}
```

Comment:
Maybe it's better to put this code into `createFormForStep` since `createForm` is exposed in the interface and might be overridden leading to the event not being dispatched.